addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -767,339 +1183,339 @@ def test_assert_created_backorders_computed_correctly(self):
-#         """Empty pickings for a picking type are unlinked if auto-unlinking is enabled."""
-#         Picking = self.env["stock.picking"]
+        # Fulfil 1st move line
+        pick.move_line_ids[0].qty_done = apple_qty_per_line
 
-#         self.picking_type_pick.u_auto_unlink_empty = True
-#         pick = self.create_picking(self.picking_type_pick)
-#         pick.u_is_empty = True
+        self.assertEqual(
+            pick.u_quantity_done,
+            apple_qty_per_line,
+            "Quantity done should match apple per line quantity",
+        )
+        self.assertTrue(pick.u_has_discrepancies, "Pick should have discrepancies")
 
-#         Picking.unlink_empty()
+        # Fulfil 2nd move line which should mean pick is no longer flagged as having discrepancies
+        pick.move_line_ids[1].qty_done = apple_qty_per_line
 
-#         self.assertFalse(pick.exists())
+        self.assertEqual(
+            pick.u_quantity_done, apple_qty, "Quantity done should match apple quantity"
+        )
+        self.assertFalse(pick.u_has_discrepancies, "Pick should not have discrepancies")
 
-#     def test_does_not_unlink_empty_pickings_if_picking_type_not_configured(self):
-#         """Empty pickings for a picking type remain if auto-unlinking is disabled."""
-#         Picking = self.env["stock.picking"]
 
-#         self.picking_type_pick.u_auto_unlink_empty = False
-#         pick = self.create_picking(self.picking_type_pick)
-#         pick.u_is_empty = True
+class TestStockPickingUoM(TestStockPickingCommon):
+    @classmethod
+    def setUpClass(cls):
+        super(TestStockPickingUoM, cls).setUpClass()
+        Uom = cls.env["uom.uom"]
+        unit = cls.env.ref("uom.product_uom_categ_unit")
+        cls.half_dozen = Uom.create(
+            {
+                "name": "HalfDozen",
+                "category_id": unit.id,
+                "factor_inv": 6,
+                "uom_type": "bigger",
+                "rounding": 1.0,
+            }
+        )
 
-#         Picking.unlink_empty()
+        cls.uom_dozen = cls.env.ref("uom.product_uom_dozen")
+        cls.cherry.uom_id = cls.uom_dozen.id
 
-#         self.assertTrue(pick.exists())
+        cls.create_quant(cls.banana.id, cls.test_stock_location_01.id, 120)
+        cls.create_quant(cls.cherry.id, cls.test_stock_location_01.id, 120)
 
+        cls.products = cls.cherry | cls.banana
 
-# class TestBatchUserName(common.BaseUDES):
-#     """Test Batch User takes value expected and changes when expected"""
+    def _create_product_infos(self, uom_id=None):
+        return [
+            {"product": self.cherry, "uom_qty": 6, "uom_id": uom_id},
+            {"product": self.banana, "uom_qty": 6, "uom_id": uom_id},
+        ]
 
-#     @classmethod
-#     def setUpClass(cls):
-#         super(TestBatchUserName, cls).setUpClass()
-#         cls.create_quant(cls.apple.id, cls.test_stock_location_01.id, 10)
+    def test_pepare_and_create_move(self):
```

Comment:
This method name and docstring could be more descriptive: what outcome are we testing?